service/src/ConstConfigIOGroup.cpp

```diff
@@ -330,18 +331,35 @@ namespace geopm
                     properties["aggregation"].string_value());
 
             auto json_values = properties["values"].array_items();
+            if (json_values.empty()) {
+                throw Exception("ConstConfigIOGroup::parse_config_json(): "
+                                "empty array of values provided for " + name,
+                                GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            if (static_cast<int>(json_values.size()) != topo.num_domain(domain_type)) {
+                throw Exception("ConstConfigIOGroup::parse_config_json(): "
```

Comment:
How will this interact with the LevelZero IOGroup if the user hasn't set the ZES_SYSMAN_ENABLE env var or when they change the number of GPUs visible using ZE_AFFFINITY_MASK? 

There could be 18 GPUs in the system and if the user sets ZE_AFINITY_MASK=16 GEOPM will only expose 1 GPU, leading to ConstConfig erroring, even though it may have all the information needed (assuming 18 values are populated).